File: petgo_app/lib/features/notify/domain/notification_item.dart
/// 通知中心条目（对应后端 `NotificationItem`，Story 6.6）。对外只有 token + deepLinkType（无顺序 id）。
class NotificationItem {
  const NotificationItem({
    required this.type,
    this.title,
    this.body,
    this.deepLinkType,
    this.deepLinkToken,
    this.targetRef,
    required this.read,
    this.createdAt,
  });

  // VET_REPLY|CONSULT_CLOSED|CONTENT_LIKED|CONTENT_COMMENTED|NEW_CONSULT_REQUEST|PET_BIRTHDAY|COMPANION_ANNIVERSARY|MILESTONE_NODE
  final String type;
  final String? title;
  final String? body;
  final String? deepLinkType;
  final String? deepLinkToken;
  /// id 寻址类跳转标识（帖子 id / 会话 id）。token 用于标记已读；targetRef 用于算目标 location。
  final String? targetRef;
  final bool read;
  final DateTime? createdAt;

  factory NotificationItem.fromJson(Map<String, dynamic> json) => NotificationItem(
        type: (json['type'] ?? '') as String,
        title: json['title'] as String?,
        body: json['body'] as String?,
        deepLinkType: json['deepLinkType'] as String?,
        deepLinkToken: json['deepLinkToken'] as String?,
        targetRef: json['targetRef'] as String?,
        read: (json['read'] ?? false) as bool,
        createdAt: json['createdAt'] == null ? null : DateTime.tryParse(json['createdAt'] as String),
      );
}

/// 通知中心分页（对应后端 `NotificationPage`）。
class NotificationPage {
  const NotificationPage({required this.items, this.nextCursor, required this.hasMore});

  final List<NotificationItem> items;
  final String? nextCursor;
  final bool hasMore;

  factory NotificationPage.fromJson(Map<String, dynamic> json) => NotificationPage(
        items: (json['items'] as List? ?? [])
            .map((e) => NotificationItem.fromJson((e as Map).cast<String, dynamic>()))
            .toList(),
        nextCursor: json['nextCursor'] as String?,
        hasMore: (json['hasMore'] ?? false) as bool,
      );
}

File: petgo-backend/src/main/java/com/tailtopia/notify/dto/NotificationItem.java
package com.tailtopia.notify.dto;

import com.tailtopia.notify.domain.Notification;
import java.time.Instant;

/**
 * 通知中心列表项（Story 6.6）。token 用于标记已读；{@code deepLinkType + targetRef} 供客户端算跳转 location。
 *
 * <p><b>targetRef 暴露说明</b>：id 寻址类通知（CONTENT_LIKED/CONTENT_COMMENTED→帖子 id、
 * VET_REPLY/CONSULT_CLOSED→会话 id）的目标页在客户端本就以数字 id 寻址（feed `/content/{id}`、
 * `/consult/conversation/{id}`），targetRef 即这些 id，未引入新的可枚举暴露；且仅下发给通知本人。
 * 固定目标类（生日/纪念日/里程碑）的 targetRef 为内部标识（如 "C-L1"），客户端不用于寻址。
 */
public record NotificationItem(
        String type,
        String title,
        String body,
        String deepLinkType,
        String deepLinkToken,
        String targetRef,
        boolean read,
        Instant createdAt) {

    public static NotificationItem from(Notification n) {
        return new NotificationItem(n.getType().name(), n.getTitle(), n.getBody(),
                n.getDeepLinkType(), n.getDeepLinkToken(), n.getTargetRef(), n.isRead(), n.getCreatedAt());
    }
}
